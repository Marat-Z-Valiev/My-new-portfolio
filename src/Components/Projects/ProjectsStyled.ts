import styled from "styled-components";

const ProjectsStyled = styled.section`
	display: flex;
	flex-direction: column;
`;

export default ProjectsStyled;
