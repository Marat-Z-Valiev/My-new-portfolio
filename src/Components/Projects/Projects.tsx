import React from "react";
import Project from "../Project/Project";
import ProjectsStyled from "./ProjectsStyled";
import projects from "../../data/projects";

const Projects: React.FC = () => {
	return (
		<ProjectsStyled>
			<div id="projects" className="projects-list">
				{Object.keys(projects).map((key) => (
					<Project key={key} project={projects[key]} />
				))}
			</div>
		</ProjectsStyled>
	);
};

export default Projects;
