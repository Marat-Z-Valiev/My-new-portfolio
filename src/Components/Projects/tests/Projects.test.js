import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Projects from "../Projects";

configure({adapter: new Adapter()});

describe("Projects", () => {
	it("should render correctly", () => {
		const wrapper = shallow(<Projects />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
