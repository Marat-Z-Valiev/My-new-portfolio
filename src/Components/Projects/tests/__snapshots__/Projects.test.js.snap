// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should render correctly 1`] = `
<ProjectsStyled>
  <div
    className="projects-list"
    id="projects"
  >
    <Project
      key="project1"
      project={
        Object {
          "backgroundClass": "white",
          "description": "The project built using ReactJS, TypeScript, React Router, Redux, Jest. Utilizing  (TMDB API). Fully responsive and accessible.",
          "githubLink": "https://github.com/Marat-Z-Valiev/movies-api-tmdb",
          "image": "https://i.ibb.co/JQr8Zmp/movie-api-image-resized.jpg",
          "name": "Movies API",
          "technologies": Array [
            "https://i.ibb.co/qYDMK54/React-js-logo.png",
            "https://i.ibb.co/QCN8KZ9/redux.png",
            "https://i.ibb.co/Ct9gHzc/typescript.png",
          ],
          "websiteLink": "https://movies-api-marat.netlify.app/",
        }
      }
    />
    <Project
      key="project2"
      project={
        Object {
          "backgroundClass": "blue",
          "description": "The project built using ReactJS, TypeScript, API, React Router, Redux, Jest",
          "githubIcon": "fa fa-github fa-2x",
          "githubLink": "https://github.com/Marat-Z-Valiev/Random_quote_machine",
          "globeIcon": "fa fa-globe fa-2x",
          "image": "https://i.ibb.co/TB1ZXJL/Pic-App-Apple-i-Mac-cropped.jpg",
          "name": "Movies API",
          "position": "right",
          "technologies": Array [
            "https://i.ibb.co/qYDMK54/React-js-logo.png",
            "https://i.ibb.co/QCN8KZ9/redux.png",
            "https://i.ibb.co/Ct9gHzc/typescript.png",
          ],
          "websiteLink": "https://marat-z-valiev.github.io/Random_quote_machine",
        }
      }
    />
    <Project
      key="project3"
      project={
        Object {
          "backgroundClass": "white",
          "description": "The project built using ReactJS, TypeScript, API, React Router, Redux, Jest",
          "githubIcon": "fa fa-github fa-2x",
          "githubLink": "https://github.com/Marat-Z-Valiev/Random_quote_machine",
          "globeIcon": "fa fa-globe fa-2x",
          "image": "https://i.ibb.co/TB1ZXJL/Pic-App-Apple-i-Mac-cropped.jpg",
          "name": "Movies API",
          "technologies": Array [
            "https://i.ibb.co/qYDMK54/React-js-logo.png",
            "https://i.ibb.co/QCN8KZ9/redux.png",
            "https://i.ibb.co/Ct9gHzc/typescript.png",
          ],
          "websiteLink": "https://marat-z-valiev.github.io/Random_quote_machine",
        }
      }
    />
  </div>
</ProjectsStyled>
`;
